### PR TITLE
Use a proper version even when not publishing

### PR DESCRIPTION
**Pull request by Luzifer**

### CODE REVIEW

File: golang.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ function step() {
 	echo "===> $@..."
 }
 
-VERSION=$(git describe --tags --exact-match || echo "ghpublish__notags")
+VERSION=$(git describe --tags || git rev-parse --short HEAD || echo "dev")
```
Comment (mlafeldt) [inline golang.sh]: I think that's exactly what `git describe --tags --always` does.

Comment (Luzifer) [reply golang.sh]: Hm you're right. Never noticed that.

**Comment by Luzifer** [comment]
PTAL again @mlafeldt

File: golang.sh [hunk 2]
```diff
@@ -78,7 +78,7 @@ fi
 
 step "Publish builds to Github"
 
-if [[ ${VERSION} == "ghpublish__notags" ]]; then
+if ! (git describe --tags --exact-match); then
```
Comment (mlafeldt) [inline golang.sh]: You don't need a subshell here. `if ! git ...` works fine.

**Review by mlafeldt — APPROVED** [review]
LGTM

**Pull request merged**